client/QuickStart/SampleApp.cs

```diff
@@ -0,0 +1,303 @@
+        /// <param name="configDatabaseName">DB name</param>
+        /// <param name="configTableName">Table name</param>
+        /// <param name="batchingPolicy">Ingestion batching policy</param>
+        private static async Task AlterBatchingPolicyAsync(ICslAdminProvider adminClient, string configDatabaseName, string configTableName, string batchingPolicy)
+        {
+            // Tip 1: Though most users should be fine with the defaults, to speed up ingestion, such as during development and in this sample app, we opt to modify the default ingestion policy to ingest data after at most 10 seconds.
+            // Tip 2: This is generally a one-time configuration.
+            // Tip 3: You can also skip the batching for some files using the Flush-Immediately property, though this option should be used with care as it is inefficient.
+            var command = $".alter table {configTableName} policy ingestionbatching @'{batchingPolicy}'";
+            await Util.ExecuteAsync(adminClient, configDatabaseName, command);
+            // If it failed to alter the ingestion policy - it could be the result of insufficient permissions. The sample will still run, though ingestion will be delayed for up to 5 minutes.
+        }
+
+        /// <summary>
+        /// Queries the data on the existing number of rows.
+        /// </summary>
+        /// <param name="queryClient">Client to run queries</param>
+        /// <param name="configDatabaseName">DB name</param>
+        /// <param name="configTableName">Table name</param>
+        private static async Task QueryExistingNumberOfRowsAsync(ICslQueryProvider queryClient, string configDatabaseName, string configTableName)
+        {
+            var query = $"{configTableName} | count";
+            await Util.ExecuteAsync(queryClient, configDatabaseName, query);
+        }
+
+        /// <summary>
+        /// Queries the first two rows of the table.
+        /// </summary>
+        /// <param name="queryClient">Client to run queries</param>
+        /// <param name="configDatabaseName">DB name</param>
+        /// <param name="configTableName">Table name</param>
+        private static async Task QueryFirstTwoRowsAsync(ICslQueryProvider queryClient, string configDatabaseName, string configTableName)
+        {
+            var query = $"{configTableName} | take 2";
+            await Util.ExecuteAsync(queryClient, configDatabaseName, query);
+        }
+
+        /// <summary>
+        /// Second phase - The ingestion process.
+        /// </summary>
+        /// <param name="config">ConfigJson object</param>
+        /// <param name="adminClient">Privileged client to run Control Commands</param>
+        /// <param name="ingestClient">Client to ingest data</param>
+        private static async Task IngestionAsync(ConfigJson config, ICslAdminProvider adminClient, IKustoIngestClient ingestClient)
+        {
+            foreach (var file in config.Data)
```

Comment:
Make this into a proper class too, instead of magic strings and parsing?

Reply:
not quite sure what you meant.  a class of what? dataFormat?

Reply:
All of the properties inside config.Data

```c#
        [JsonObject(NamingStrategyType = typeof(CamelCaseNamingStrategy))]

        class FileData
        {
            private string Format { get; set; };
            ...
        }
```

Reply:
If you'll embed it in ConfigJson (`public List<ConfigData> Data { get; set; }`), it will parse it automatically when you read the config.